app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -302,6 +253,26 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
         super.onNewIntent(unsafeIntent)
     }
 
+    private fun handleAppNavigation(intent: SafeIntent) {
+        if (components.appStore.state.screen == Screen.Locked()) {
+            components.appStore.dispatch(AppAction.Lock(intent.extras))
+            lifecycle.addObserver(navigator)
+            return
+        }
+        SearchWidgetNavigation.handleAppNavigation(
+            intent.extras,
+            this,
+            splashScreenPreDrawListener = {
+                setSplashScreenPreDrawListener(intent)
+            },
+            navigator = {
+                lifecycle.addObserver(navigator)
+            },
+            unlock = {
+            },
```

Comment:
Can defaults be provided in `handleAppNavigation` for the parameters which are not mandatory?

Reply:
Done